kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -175,23 +181,50 @@ public static Map<String, Variant<?>> build80211WirelessSettings(NetworkProperti
     }
 
     public static Map<String, Variant<?>> build80211WirelessSecuritySettings(NetworkProperties props, String deviceId) {
+        String propMode = props.get(String.class, KURA_PROPS_KEY_WIFI_MODE, deviceId);
+        String securityType = props.get(String.class, KURA_PROPS_KEY_WIFI_SECURITY_TYPE, deviceId,
+                propMode.toLowerCase());
+
+        if ("SECURITY_WEP".equals(securityType)) {
```

Comment:
I think you're correct. Adding a small `KuraWifiSecurityType` enum that represents the possible values we can find in the snapshot would be a better option than comparing Strings. This way it would be less error-prone, more type safe and cleaner.

If I have time I'll try to submit a new PR to refactor this.